nomad/job_endpoint_hooks.go

```diff
@@ -120,11 +131,17 @@ func (jobImpliedConstraints) Mutate(j *structs.Job) (*structs.Job, []error, erro
 	// Get the required signals
 	signals := j.RequiredSignals()
 
+	// Identify which task groups are utilising Nomad native service discovery.
+	nativeServiceDisco := j.RequiredNativeServiceDiscovery()
+
 	// Hot path
-	if len(signals) == 0 && len(policies) == 0 {
+	if len(signals) == 0 && len(policies) == 0 && len(nativeServiceDisco) == 0 {
 		return j, nil, nil
 	}
 
+	// TODO(jrasell) can we perform the constraint checking and additions in a
+	//  single task group loop, rather than one per constraint. Investigate.
+
```

Comment:
I'd say either fix it now (single loop over task groups with a function call per constraint I think), or remove the TODO. If it's not worth doing while you're thinking about it, it's probably not going to happen. :sweat_smile:

Reply:
OK, I'll remove the comment at the compromise of raising an issue and assigning it to myself. I'd like to fix it, but not within these commits to avoid touching too much in what is already a large piece of work.